src/helics/core/tcp/TcpHelperClasses.cpp

```diff
@@ -37,8 +37,9 @@ namespace tcp {
             if (!triggerhalt) {
                 socket_.async_receive(
                     asio::buffer(data.data() + residBufferSize, data.size() - residBufferSize),
-                    [this](const std::error_code& error, size_t bytes_transferred) {
-                        handle_read(error, bytes_transferred);
+                    [ptr = shared_from_this()](
+                        const std::error_code& error, size_t bytes_transferred) {
```

Comment:
Slightly wrapping. Is this from the line length?

Reply:
the formatting will automatically wrap from the line length, though I think the algorithm is some penalty function so in some cases it can be slightly longer.    You think this is an issue that we need to tweak?

Reply:
I think in most other cases it would look fine, it's just with the lambda that it looks a bit strange because the lambda body is indented to the same level as the args.

Reply:
I have noticed different version of clang-format do handle lambdas a little different.  Do you think it is something we should tweak.  Changing the variable names can probably shift it around a little if you think it worthwhile.

Reply:
Maybe try a variable name change. If it doesn't work then it's okay, most of the other lambdas I saw didn't wrap like this one.